common/src/com/thoughtworks/go/config/materials/git/GitMaterial.java

```diff
@@ -106,28 +127,32 @@ public MaterialConfig config() {
     }
 
     public List<Modification> modificationsSince(File baseDir, Revision revision, final SubprocessExecutionContext execCtx) {
+        unshallowIfNeeded(ProcessOutputStreamConsumer.inMemoryConsumer(), revision, baseDir);
         GitCommand gitCommand = getGit(baseDir);
         return gitCommand.modificationsSince(revision);
     }
 
     public MaterialInstance createMaterialInstance() {
-        return new GitMaterialInstance(url.forCommandline(), branch, submoduleFolder, UUID.randomUUID().toString());
+        return new GitMaterialInstance(url.forCommandline(), branch, submoduleFolder, UUID.randomUUID().toString(), shallowClone);
     }
 
     @Override
     protected void appendCriteria(Map<String, Object> parameters) {
         parameters.put(ScmMaterialConfig.URL, url.forCommandline());
         parameters.put("branch", branch);
+        parameters.put("shallowClone", shallowClone);
```

Comment:
Umm, that might not be ok. Changing fingerprint is not something that should be done as far as possible, since it has quite big implications on fan-in, etc. Might not show easily during an upgrade. This needs @jyotisingh's expertise. CCing her.

Does it really need to part of the fingerprint? Can't it just be stored in the DB, without being part of the fingerprint? Anyway, @jyotisingh will know more/better.